dipy/io/utils.py

```diff
@@ -136,6 +138,80 @@ def decfa_to_float(img_orig):
     return Nifti1Image(out_data, affine=img_orig.affine, header=new_hdr)
 
 
+def is_reference_info_valid(affine, dimensions, voxel_sizes, voxel_order):
+    """ Will validate basic data type and value of spatial attribute.
+    Does not ensure that voxel_sizes and voxel_order are self-coherent with
+    affine.
+    The listed parameters are what is expected, provide something else and this
+    function should fail (cover common mistakes).
+
+    Parameters
+    ----------
+        affine: ndarray (4,4)
+            Tranformation of VOX to RASMM
+        dimensions: ndarray (3,), int16
+            Volume shape for each axis
+        voxel_sizes:  ndarray (3,), float32
+            Size of voxel for each axis
+        voxel_order: string
+            Typically 'RAS' or 'LPS'
+
+    Returns
+    -------
+    output : bool
+        Does the input represent a valid 'state' of spatial attribute
+    """
+    all_valid = True
+    only_3d_warning = False
+
+    if not affine.shape == (4, 4):
+        all_valid = False
+        logging.warning('Transformation matrix must be 4x4')
+
+    if not affine[0:3, 0:3].any():
+        all_valid = False
+        logging.warning('Rotation matrix cannot be all zeros')
+
+    if not len(dimensions) >= 3:
+        all_valid = False
+        only_3d_warning = True
+
+    for i in dimensions:
+        if not isinstance(i, numbers.Integral):
+            all_valid = False
+            logging.warning('Dimensions must be int.')
+        if i <= 0:
+            all_valid = False
+            logging.warning('Dimensions must be above 0.')
+
+    if not len(voxel_sizes) >= 3:
+        all_valid = False
+        only_3d_warning = True
+    for i in voxel_sizes:
+        if not isinstance(i, numbers.Number):
+            all_valid = False
+            logging.warning('Voxel size must be int/float.')
+        if i <= 0:
+            all_valid = False
+            logging.warning('Voxel size must be above 0.')
+
+    if not len(voxel_order) >= 3:
+        all_valid = False
+        only_3d_warning = True
+    for i in voxel_order:
+        if not isinstance(i, six.string_types):
```

Comment:
In Python 3, I believe that this could be simply: 

```suggestion
        if not isinstance(i, str):
```